flow/cmd/custom_sync.go

```diff
@@ -0,0 +1,171 @@
+package cmd
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/peerdbenv"
+	peerflow "github.com/PeerDB-io/peer-flow/workflows"
+
+	"crypto/sha256"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const peerdbPauseGuideDocLink = "https://docs.peerdb.io/features/pause-mirror"
+
+func AuthenticateSyncRequest(ctx context.Context) error {
```

Comment:
should be implemented using gRPC middleware ideally: https://shijuvar.medium.com/writing-grpc-interceptors-in-go-bf3e7671fe48

Reply:
we will be exposing these as http endpoints as well